packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -4,33 +4,70 @@ import traverse, { Scope, NodePath } from '@babel/traverse';
 import { hash, unique } from '@compiled/utils';
 import { transformCss } from '@compiled/css';
 import isPropValid from '@emotion/is-prop-valid';
-
 import { CSSOutput } from './css-builders';
 import { PluginOptions } from '../types';
 
 interface BaseOpts extends PluginOptions {
+  /**
+   * CSS data that will be integrated into the output AST.
+   */
   cssOutput: CSSOutput;
 }
 
 interface StyledOpts extends BaseOpts {
+  /**
+   * Tag of the Styled Component,
+   * for example `"div"`.
+   */
   tagName: string;
+
+  /**
+   * Babel path used for traversing inner nodes.
+   */
   parentPath: NodePath;
+
+  /**
+   * Babel scope used for traversing inner nodes.
+   */
   scope: Scope;
 }
 
 interface CompiledOpts extends BaseOpts {
   node: t.JSXElement;
 }
 
+/**
+ * Will build up the CSS variables prop to be placed as inline styles.
+ *
+ * @param variables CSS variables that will be placed in the AST
+ * @param transform Transform function that can be used to change the CSS variable expression
+ */
 const buildCssVariablesProp = (
   variables: CSSOutput['variables'],
   transform = (expression: t.Expression) => expression
 ): (t.ObjectProperty | t.SpreadElement)[] => {
-  return unique(variables, (item) => item.name).map((variable) => {
-    return t.objectProperty(t.stringLiteral(variable.name), transform(variable.expression));
+  return unique(
+    // Make sure all defined CSS variables are unique
+    variables,
+    // We consider their uniqueness based on their name
```

Comment:
Is this enough to consider them unique? It's valid css to have two variables with the same name in the same scope, the cascade just takes over and the last one defined is used. If we use a `find` for example, this could return something other than what the user intended

Reply:
in this case its fine because it will always be the closest name.

```
import '@compiled/core';

let color = 'black';

export default function Component () {
  let color = 'red';
  return <div css={{ color }} />
}
```

99% case last reference wins - which is OK i think?